### DISCUSSION TITLE
ReactJS

### DISCUSSION THREAD

**Discussion by MK-Hasan-Mahdi** [MK-Hasan-Mahdi/jspractice#2] [Q&A] [answered]
How ReactJS works?

**Comment by ImRanKhan81m** [comment, marked as answer]
ReactJS divides the UI into isolated reusable pieces of code known as components. React components work similarly to JavaScript functions as they accept arbitrary inputs called properties or props. It's possible to have as many components as necessary without cluttering your code.

**Comment by AyonJD** [comment]
React. js is an open-source JavaScript library that is used for building user interfaces specifically for single-page applications. It's used for handling the view layer for web and mobile apps. React also allows us to create reusable UI components.

**Comment by AyonJD** [comment]
React. js is an open-source JavaScript library that is used for building user interfaces specifically for single-page applications. It's used for handling the view layer for web and mobile apps. React also allows us to create reusable UI components.
